File: 01_A/VB/Application_1/Ball.vb
Public Class Ball
    Public x As Integer
    Public y As Integer
    Public radius As Integer
    Public moveX As Integer
    Public moveY As Integer

    Public Sub New(ByVal x As Integer, ByVal y As Integer, ByVal radius As Integer, ByVal moveX As Integer, ByVal moveY As Integer)
        Me.x = x
        Me.y = y
        Me.radius = radius
        Me.moveX = moveX
        Me.moveY = moveY
    End Sub
    Public Sub MoveBall(ByVal box As Rectangle)
        x += moveX

        If x + radius >= box.Width OrElse x <= box.X Then
            moveX = -moveX
        End If

        y += moveY

        If y + radius >= box.Height OrElse y <= box.Y Then
            moveY = -moveY
        End If
    End Sub

End Class

File: 01_A/C#/Application_1/Ball.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;

namespace Application_1
{
    // Creating a separate class for the balls to manage multiple instances
    class Ball
    {
        public int x;
        public int y;
        public int radius;
        public int moveX;
        public int moveY;

        public Ball(int x, int y, int radius, int moveX, int moveY)
        {
            this.x = x;
            this.y = y;
            this.radius = radius;
            this.moveX = moveX;
            this.moveY = moveY;
        }

        public void MoveBall(Rectangle box)
        {
            x += moveX;
            if ((x+radius) >= box.Width || x <= box.X)
            {
                moveX = -moveX;
            }
            y += moveY;
            if ((y + radius) >= box.Height || y <= box.Y)
            {
                moveY = -moveY;
            }
        }

    }
}
